pkg/cmd/container/cp.go

```diff
@@ -0,0 +1,277 @@
+	srcSpec, err := parseCpFileSpec(args[0])
+	if err != nil {
+		return err
+	}
+
+	destSpec, err := parseCpFileSpec(args[1])
+	if err != nil {
+		return err
+	}
+
+	if (srcSpec.Container != nil && destSpec.Container != nil) || (len(srcSpec.Path) == 0 && len(destSpec.Path) == 0) {
+		return fmt.Errorf("one of src or dest must be a local file specification")
+	}
+	if srcSpec.Container == nil && destSpec.Container == nil {
+		return fmt.Errorf("one of src or dest must be a container file specification")
+	}
+	if srcSpec.Path == "-" {
+		return fmt.Errorf("support for reading a tar archive from stdin is not implemented yet")
+	}
+	if destSpec.Path == "-" {
+		return fmt.Errorf("support for writing a tar archive to stdout is not implemented yet")
+	}
+
+	container2host := srcSpec.Container != nil
+	var container string
+	if container2host {
+		container = *srcSpec.Container
+	} else {
+		container = *destSpec.Container
+	}
+
+	// cp works in the host namespace (for inspecting file permissions), so we can't directly use the Go client.
+	inspectArgs := append(options.NerdctlArgs, "container", "inspect", "--mode=native", "--format={{json .Process}}", container)
+	inspectCmd := exec.CommandContext(ctx, options.NerdctlCmd, inspectArgs...)
+	inspectCmd.Stderr = os.Stderr
+	inspectOut, err := inspectCmd.Output()
+	if err != nil {
+		return fmt.Errorf("failed to execute %v: %w", inspectCmd.Args, err)
+	}
+	var proc native.Process
+	if err := json.Unmarshal(inspectOut, &proc); err != nil {
+		return err
+	}
+	if proc.Status.Status != containerd.Running {
+		return fmt.Errorf("expected container status %v, got %v", containerd.Running, proc.Status.Status)
+	}
+	if proc.Pid <= 0 {
+		return fmt.Errorf("got non-positive PID %v", proc.Pid)
+	}
+	return kopy(ctx, container2host, proc.Pid, destSpec.Path, srcSpec.Path, options.FollowLink)
+}
+
+var errFileSpecDoesntMatchFormat = errors.New("filespec must match the canonical format: [container:]file/path")
+
+func parseCpFileSpec(arg string) (*cpFileSpec, error) {
+	i := strings.Index(arg, ":")
+
+	// filespec starting with a semicolon is invalid
+	if i == 0 {
+		return nil, errFileSpecDoesntMatchFormat
+	}
+
+	if filepath.IsAbs(arg) {
+		// Explicit local absolute path, e.g., `C:\foo` or `/foo`.
+		return &cpFileSpec{
+			Container: nil,
+			Path:      arg,
+		}, nil
+	}
+
+	parts := strings.SplitN(arg, ":", 2)
+
+	if len(parts) == 1 || strings.HasPrefix(parts[0], ".") {
+		// Either there's no `:` in the arg
+		// OR it's an explicit local relative path like `./file:name.txt`.
+		return &cpFileSpec{
+			Path: arg,
+		}, nil
+	}
+
+	return &cpFileSpec{
+		Container: &parts[0],
+		Path:      parts[1],
+	}, nil
+}
+
+type cpFileSpec struct {
+	Container *string
+	Path      string
+}
+
+// kopy implements `nerdctl cp`.
+//
+// See https://docs.docker.com/engine/reference/commandline/cp/ for the specification.
+func kopy(ctx context.Context, container2host bool, pid int, dst, src string, followSymlink bool) error {
```

Comment:
Should be moved to `pkg/containerutil.Cp`

Reply:
Got it
Done~